runtime/lib/Handlers/setV3/lib/evaluateScript.ts

```diff
@@ -0,0 +1,60 @@
+import { CompiledSetV3NodeDataItem, SetV3NodeDataItemType } from '@voiceflow/dtos';
+import _ from 'lodash';
+
+import Store from '@/runtime/lib/Runtime/Store';
+import { toErrorReasonString } from '@/utils/error/toErrorReasonString';
+import { joinTokens } from '@/utils/string/joinTokens';
+
+import { IVMIsolate } from '../../utils/ivmIsolate';
+import { SetEvaluator } from './setEvaluator';
+import { getSetCode, getSetVariable } from './utils';
+
+async function evaluate(
+  isolate: IVMIsolate,
+  item: CompiledSetV3NodeDataItem,
+  updatePatch: Record<string, unknown>,
+  logToPrototype: (message: string) => void
+) {
+  try {
+    if (item.type !== SetV3NodeDataItemType.SCRIPT) {
+      throw new Error(`received the wrong type of set assignment, received type '${item.type}'`);
+    }
+
+    if (!item.value) {
+      throw new Error(`code is undefined`);
+    }
+
+    if (!item.variable) {
+      throw new Error(`target variable is undefined`);
+    }
+
+    const value = await isolate.executeFunctionOrScript(item.value);
+    updatePatch[item.variable] = value;
+  } catch (err) {
+    const prefix = joinTokens([`Expression`, getSetCode(item), 'could not set', getSetVariable(item)]);
+    const suffix = toErrorReasonString(err);
+    const errMessage = joinTokens([prefix, suffix]);
+
+    logToPrototype(errMessage);
+  }
+}
+
+export const evaluateScript: SetEvaluator = async (
+  item: CompiledSetV3NodeDataItem,
+  variables: Store,
+  logToPrototype: (message: string) => void
+): Promise<Record<string, unknown>> => {
+  const isolate = new IVMIsolate(variables.getState());
+
+  try {
+    const updatePatch: Record<string, unknown> = {};
+
+    await isolate.initialize();
+
+    await evaluate(isolate, item, updatePatch, logToPrototype);
+
+    return updatePatch;
```

Comment:
```suggestion
    await isolate.initialize();

    return await evaluate(isolate, item, updatePatch, logToPrototype);
```
passing by reference and mutating adds unnecessary complexity. `evaluate` can be in charge of creating and returning the `updatePatch`

Reply:
Leftover pattern from a previous iteration where this was more useful. Removed.